### PR TITLE
README.md: fix nits

### PR DESCRIPTION
**Pull request by demonCoder95**
Fixed many broken links in the doc. Most importantly, the Client
API doc no longer exists, so I pointed that to the Kubebuilder
book, since that provides an interesting insight into Reconcilers,
 Controller, Events and such.

### CODE REVIEW

**Comment by demonCoder95** [comment]
This PR will also close issue #74

File: README.md [hunk 1]
```diff
@@ -469,25 +471,25 @@ Hopefully, this guide was an effective demonstration of the value of the Operato
...
 [doc_olm_install_instr]: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/install/install.md
-[layout_doc]: https://github.com/operator-framework/operator-sdk/blob/master/doc/project_layout.md
+[layout_doc]: https://sdk.operatorframework.io/docs/golang/references/project-layout/
 [manager_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/manager#Manager
 [controller_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller
 [memcached_controller]: https://github.com/operator-framework/operator-sdk/blob/master/example/memcached-operator/memcached_controller.go.tmpl
 [request_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/reconcile#Request
 [result_go_doc]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/reconcile#Result
-[doc_client_api]: https://github.com/operator-framework/operator-sdk/blob/master/doc/user/client.md
+[doc_kubebuilder_book]: https://book-v1.book.kubebuilder.io/
```
Comment (camilamacedo86) [inline README.md:483]: The client API doc should be : https://sdk.operatorframework.io/docs/golang/references/client/

File: README.md [hunk 2]
```diff
@@ -178,7 +178,7 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 }  
 ```
 
-For a guide on Reconcilers, Clients, and interacting with resource Events, see the [Client API doc][doc_client_api].
+For a guide on Reconcilers, Clients, and interacting with resource Events, see the [Kubebuilder Book][doc_kubebuilder_book].
```
Comment (camilamacedo86) [inline README.md:181]: Revert this one and point out to https://sdk.operatorframework.io/docs/golang/references/client/

File: README.md (hunk 1 above)
Comment (demonCoder95) [reply README.md:483]: Fixed. Sending a new commit.

**Comment by demonCoder95** [comment]
I think the job failed randomly, the LCM link is not broken. You can recheck the jobs.

**Comment by estroz** [comment]
Closes #73

**Comment by estroz** [comment]
@demonCoder95 our markdown checker is having issues. Will force merge.

**Comment by demonCoder95** [comment]
Oh, okay. No problem. Happy to help.

**Comment by estroz** [comment]
@demonCoder95 thanks for contributing!

**Comment by demonCoder95** [comment]
No problem! I see broken links and typos all the time. I think taking a couple minutes out to fix it doesn't hurt. Also, I love when a community responds quickly and merges the changes, helps me stay inclined to contribute more!

**Comment by demonCoder95** [comment]
Also, you'll probably want to close issues #73  and #74  manually now, since the merged was forced.

**Pull request merged**